spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/function/FunctionDetectorCondition.java

```diff
@@ -101,11 +110,31 @@ public ConditionOutcome getMatchOutcome(ConditionContext context, AnnotatedTypeM
 						prunedList.add(key);
 					}
 				}
+				else {
+					//check if its a @Component bean.
+					Optional<Method> componentBeanMethod = Arrays.stream(methods).filter(
+							m -> m.getName().equals("apply") && isKafkaStreamsTypeFound(m) ||
+									m.getName().equals("accept") && isKafkaStreamsTypeFound(m)).findFirst();
+					if (componentBeanMethod.isPresent()) {
+						Method method = componentBeanMethod.get();
+						final ResolvableType resolvableType1 = ResolvableType.forMethodParameter(method, 0);
+						final Class<?> rawClass = resolvableType1.getRawClass();
+						if (rawClass == KStream.class || rawClass == KTable.class || rawClass == GlobalKTable.class) {
+							prunedList.add(key);
+						}
+					}
+				}
 			}
 			catch (Exception e) {
 				LOG.error("Function not found: " + key, e);
```

Comment:
Just log (and continue) or may be fail fast? If function is not found can the app function?

Reply:
It is ok to simply log and continue, because we may have other types of functions.